packages/jest-haste-map/src/index.ts

```diff
@@ -128,7 +128,7 @@ const MAX_WAIT_TIME = 240000;
 const NODE_MODULES = `${path.sep}node_modules${path.sep}`;
 const PACKAGE_JSON = `${path.sep}package.json`;
 const VCS_DIRECTORIES = ['.git', '.hg', '.sl']
-  .map(vcs => escapePathForRegex(path.sep + vcs + path.sep))
+  .map(vcs => escapePathForRegex(`/${vcs}/`))
```

Comment:
should `NODE_MODULES` and `PACKAGE_JSON` above also use forward slashes?

Reply:
These patterns is just a part of the problem. It appears that on Windows `watchPathIgnorePatterns` does not work as well.

These strings are used as paths mostly (hence `path.sep` is used), but at some point they are passed to `anymatch` which expects globs. So, I think the best would be to normalise them before passing to `anymatch`.

See the discussion starting from https://github.com/jestjs/jest/issues/13869#issuecomment-1503848263